client/src/pages/course/task/[taskDetails].tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useMemo, useState } from 'react';
+import { Header } from 'components';
+import { useRouter } from 'next/router'
+import { useAsync } from 'react-use';
+import {
+    Row,
+    Col,
+    Typography,
+    Tooltip,
+} from 'antd';
+import { EventTypeColor, EventTypeToName } from 'components/Schedule/model';
+import {
+    renderTag,
+} from 'components/Table';
+import withCourseData from 'components/withCourseData';
+import { withSession } from 'components';
+import { useLoading } from 'components/useLoading';
+import { CourseService, CourseTaskDetails } from '../../../services/course';
+import { CoursePageProps } from 'services/models';
+
+export function TaskDetailsPage(props: CoursePageProps) {
+    const router = useRouter();
+    const { taskDetails } = router.query;
+
+    const [taskData, setTaskData] = useState<CourseTaskDetails>();
+    const [, withLoading] = useLoading(false);
+
+    const courseService = useMemo(() => new CourseService(props.course.id), [props.course.id]);
+
+    useAsync(
+        withLoading(async () => {
+            const task = await courseService.getCourseTask(taskDetails as string);
+            setTaskData(task as CourseTaskDetails);
+        }),
+        [courseService],
+    );
+    const { Title, Text } = Typography;
```

Comment:
We use frameworks to reuse code as much as possible. Task details and  event details components have pretty same logic, but different views. Let's use one of the React technics to avoid duplications.

Reply:
done